internal/manifest/v1beta1/ready_check.go

```diff
@@ -56,3 +71,36 @@ func (c *ManifestCustomResourceReadyCheck) Run(
 
 	return nil
 }
+
+func checkDeploymentState(resources []*resource.Info) (bool, error) {
+	var deploy *appsv1.Deployment
+	for _, res := range resources {
+		typedObject := kube.AsVersioned(res)
```

Comment:
this is a helm dependency we do not need I think. IF you bind the GVK anyway against appsv1, you can use the typedObject directly with the default converter from the scheme